packages/perseus-editor/src/widgets/free-response-editor.tsx

```diff
@@ -108,6 +111,37 @@ class FreeResponseEditor extends React.Component<Props> {
                         }
                     />
                 </label>
+                <label
+                    className={css(styles.textOptionWithLabelContainer)}
+                    htmlFor="allow-unlimited-characters"
+                >
+                    <HeadingSmall>Allow unlimited characters</HeadingSmall>
+                    <Checkbox
+                        checked={this.props.allowUnlimitedCharacters}
+                        id="allow-unlimited-characters"
+                        onChange={(val) =>
+                            this.props.onChange({
+                                allowUnlimitedCharacters: val,
+                            })
+                        }
+                    />
+                </label>
+                {!this.props.allowUnlimitedCharacters && (
+                    <label className={css(styles.textOptionWithLabelContainer)}>
+                        <HeadingSmall>Character limit</HeadingSmall>
+                        <input
+                            type="number"
+                            min={1}
+                            value={this.props.characterLimit}
+                            onChange={(e) => {
+                                const val = parseInt(e.target.value);
```

Comment:
I think we need to check if `val` is not `NaN`, and either not process the change or call `props.onChange` with `1`. If we don't do that, we'll pass through `NaN` to React for rendering, which will cause errors.